### ISSUE TITLE
Why only openmpi is supported with horovodrun?

### ISSUE THREAD

**Issue by denfromufa** [horovod/horovod#1164]
Why only openmpi is supported with horovodrun? For example why not also support Spectrum MPI which is based on openmpi and should be mostly compatible.

```
(powerai.1.6.1) [@apps]$ horovodrun -np 4 python keras_mnist.py
Open MPI not found in output of mpirun --version.
Traceback (most recent call last):
  File "/data/gpfs/Users/j05/anaconda3/envs/powerai.1.6.1/bin/horovodrun", line 21, in <module>
    run.run()
  File "/data/gpfs/Users/j05/anaconda3/envs/powerai.1.6.1/lib/python3.6/site-packages/horovod/run/run.py", line 448, in run
    'horovodrun convenience script currently only supports '
Exception: horovodrun convenience script currently only supports Open MPI.

Choose one of:
1. Install Open MPI 4.0.0+ and re-install Horovod (use --no-cache-dir pip option).
2. Run distributed training script using the standard way provided by your MPI distribution (usually mpirun, srun, or jsrun).

```

**Comment by alsrgv** [comment]
@denfromufa, would you be willing to send a PR for and maintain Spectrum MPI support?

**Comment by denfromufa** [comment]
@alsrgv I read this document below and although smpi is based on openmpi and emulates the core APIs, still I think there are some differences in arguments used by Horovod. For example, the way openib & btl are specified.  Anyway I would probably be NOT the right person to support this. Maybe @nvcastet who merged the DDL support to Horovod :)

https://github.com/horovod/horovod/blob/master/docs/mpirun.rst

The use case is for benchmarks and for switching between openmpi and smpi on various machines or environments, without worrying about the differences in mpirun arguments that would be too overwhelming to end-users.

**Comment by alsrgv** [comment]
I think it’d be great to have Spectrum MPI support. @nvcastet are you up for adding support to horovodrun and horovod.spark.run (for completeness)?

**Comment by nvcastet** [comment]
@alsrgv @denfromufa Thank you for bring that up. I will talk with my team about it to see what we can do.

**Comment by tgaddair** [comment]
Fixed by #1441 for Spectrum MPI.  Gloo is also now supported by #1181.

**Comment by shailensobhee** [comment]
What about Intel MPI?

Output of `horovodrun`:

```
[...]
Extension tensorflow was built with MPI.
Open MPI/Spectrum MPI/MPICH not found in output of mpirun --version.
[...]
```

```
mpirun -V
Intel(R) MPI Library for Linux* OS, Version 2019 Update 6 Build 20191024 (id: 082ae5608)
Copyright 2003-2019, Intel Corporation.
```
